modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractJavaCodegen.java

```diff
@@ -2492,4 +2537,40 @@ public static void addImports(List<Map<String, String>> imports, CodegenModel cm
             imports.add(importsItem);
         }
     }
+
+    /**
+     * Lambda to help the generation of {@literal @}JsontypeName.
+     *
+     *
+     * <p>
+     *     The main use case is to set the correct @jsonTypeName when inheritance is specified.
+     * <p>
+     *     No generated class is declared final, so @JsontypeName is also added for simple pojos.
+     * <p>
+     *
+     * {@literal @}JsontypeName is typically needed when
+     * <ul>
+     *   <li>modelNameSuffix and/or modelNamePrefix are used
+     *   <li>the model name is in lowercase
+     *   <li>the class name is sanitized (for example the name contains dots)
+     *   <li>x-discriminator-value is specified on the child model
+     *   <li>discriminator mapping on the parent model specifies a name different than the class name
+     *  </ul>
+     */
+    private static class UseJsonTypeName implements Mustache.Lambda {
+
+        /**
+         * Write the enclosed fragment if the json type name is different than the class name.
+         */
+        @Override
+        public void execute(Template.Fragment fragment, Writer writer) throws IOException {
+            // context index 1 is only valid when used in pojo.mustache
+            CodegenModel codegenModel = (CodegenModel)fragment.context(1);
```

Comment:
Updating samples via `./bin/generate-samples.sh ./bin/configs/*.yaml` generates an error. The reason is that the cast in this line doesn't work in one of the sample cases.
> org.openapitools.codegen.model.ModelMap cannot be cast to class org.openapitools.codegen.CodegenModel

Reply:
Sorry my mistake.  I used an obsolete version of the cli to generate the samples.
I' ve regenerated them with a fix for the cast.